src/RecoveryServices/RecoveryServices.Backup.Models/CommonModels/PolicyRetentionObjects.cs

```diff
@@ -308,17 +358,26 @@ public class DailyRetentionSchedule : RetentionScheduleBase
         public int DurationCountInDays { get; set; }
 
         // no extra fields
-        public override void Validate(ScheduleRunType ScheduleRunFrequency = 0)
+        public void Validate(ScheduleRunType ScheduleRunFrequency = 0, int MinDuration = 0, int MaxDuration = 0)
         {
             int MinDurationCountInDays = 7, MaxDurationCountInDays = PolicyConstants.MaxAllowedRetentionDurationCount;
             if(BackupManagementType == Management.RecoveryServices.Backup.Models.BackupManagementType.AzureStorage)
             {
-                MinDurationCountInDays = PolicyConstants.AfsDailyRetentionDaysMin;
-                MaxDurationCountInDays = PolicyConstants.AfsVaultDailyRetentionDaysMax;                
+                if (MinDuration != 0 && MaxDuration != 0)
+                {
+                    MinDurationCountInDays = MinDuration;
+                    MaxDurationCountInDays = MaxDuration;
+                }
+                else
+                {
+                    MinDurationCountInDays = PolicyConstants.AfsDailyRetentionDaysMin;
+                    MaxDurationCountInDays = PolicyConstants.AfsDailyRetentionDaysMax;
```

Comment:
why do we set default min max values to AFS min-max ?

Reply:
@hiaga in-order to give default values to AFS min/max in case the BackupManagementType is AzureStorage & we haven't passed in optional params for min & max. For vault policy, we pass in our custom params, however we snapshot retention for AFS it will default to these constants, matching the portal experience.